x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings.test.tsx

```diff
@@ -7,31 +7,47 @@
 
 import '../../../__mocks__/shallow_useeffect.mock';
 
-import { setMockActions } from '../../../__mocks__';
+import { setMockValues, setMockActions } from '../../../__mocks__';
 
 import React from 'react';
 
 import { shallow } from 'enzyme';
 
+import { QueryPerformance } from './query_performance';
 import { ResultSettings } from './result_settings';
 import { ResultSettingsTable } from './result_settings_table';
 
 describe('RelevanceTuning', () => {
+  const values = {
+    dataLoading: false,
+  };
+
   const actions = {
     initializeResultSettingsData: jest.fn(),
   };
+
   beforeEach(() => {
+    setMockValues(values);
     setMockActions(actions);
     jest.clearAllMocks();
   });
 
   it('renders', () => {
     const wrapper = shallow(<ResultSettings engineBreadcrumb={['test']} />);
     expect(wrapper.find(ResultSettingsTable).exists()).toBe(true);
+    expect(wrapper.find(QueryPerformance).exists()).toBe(true);
   });
 
   it('initializes result settings data when mounted', () => {
     shallow(<ResultSettings engineBreadcrumb={['test']} />);
     expect(actions.initializeResultSettingsData).toHaveBeenCalled();
   });
+
+  it('does not show a query performance rating when data is still loading', () => {
+    setMockValues({
+      dataLoading: true,
+    });
+    const wrapper = shallow(<ResultSettings engineBreadcrumb={['test']} />);
+    expect(wrapper.find(QueryPerformance).exists()).toBe(false);
+  });
```

Comment:
Can we simplify this by doing an early `if (dataLoading) return <Loading />;` return like our other views?

Reply:
Alternatively, if you'd rather make that change in your next general "Chrome" PR I'd be fine with that.

Reply:
Nope, I just handled this when I resolved the merge conflict.

Reply:
https://github.com/elastic/kibana/pull/96230/commits/a546b4c1a6a73c109a0c259245bd6419c982fe5f